arcgis-ios-sdk-samples/Layers/Style WMS layers/StyleWebMapServiceLayerViewController.swift

```diff
@@ -0,0 +1,107 @@
+    /// The map view managed by the view controller.
+    @IBOutlet weak var mapView: AGSMapView!
+    @IBOutlet weak var segmentedControl: UISegmentedControl!
+    
+    required init?(coder aDecoder: NSCoder) {
+        // Create the map.
+        map = AGSMap()
+        
+        // Create the WMS layer and add it to the map.
+        layer = AGSWMSLayer(
+            url: URL(string: "http://geoint.lmic.state.mn.us/cgi-bin/wms?VERSION=1.3.0&SERVICE=WMS&REQUEST=GetCapabilities")!,
+            layerNames: ["fsa2017"]
+        )
+        map.operationalLayers.add(layer)
+        
+        super.init(coder: aDecoder)
+        
+        // Load the WMS layer.
+        layer.load { [weak self] (error) in
+            if let error = error {
+                self?.layerDidFailToLoad(with: error)
+            } else {
+                self?.layerDidLoad()
+            }
+        }
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        (self.navigationItem.rightBarButtonItem as! SourceCodeBarButtonItem).filenames = ["StyleWebMapServiceLayerViewController"]
+        
+        // Assign the map to the map view.
+        mapView.map = map
+        
+        // The segmented control needs to be enabled here if the layer is
+        // already loaded.
+        enableSegmentedControlIfPosasible()
+    }
+    
+    /// Called in response to the layer loading successfully.
+    func layerDidLoad() {
+        guard let sublayer = layer.sublayers.firstObject as? AGSWMSSublayer else {
+            return
+        }
+        styles = sublayer.sublayerInfo.styles
+        // The segmented control needs to be enabled here if the view is already
+        // loaded.
+        enableSegmentedControlIfPosasible()
+    }
+    
+    /// Called in response to the layer failing to load. Presents an alert
+    /// announcing the failure.
+    ///
+    /// - Parameter error: The error that caused loading to fail.
+    func layerDidFailToLoad(with error: Error) {
+        let alertController = UIAlertController(title: nil, message: "Failed to load WMS layer", preferredStyle: .alert)
+        let okayAction = UIAlertAction(title: "OK", style: .default)
+        alertController.addAction(okayAction)
+        alertController.preferredAction = okayAction
+        present(alertController, animated: true)
+    }
+    
+    /// Enables the segmented control if the view has loaded and there is more
+    /// than one style.
+    func enableSegmentedControlIfPosasible() {
+        guard isViewLoaded, styles.count > 1 else { return }
+        segmentedControl.isEnabled = true
+    }
+    
+    @IBAction func changeStyle(_ sender: UISegmentedControl) {
+        guard let sublayer = layer.sublayers.firstObject as? AGSWMSSublayer else {
+            preconditionFailure()
+        }
+        sublayer.currentStyle = styles[sender.selectedSegmentIndex]
+    }
+    
+    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
```

Comment:
Is this override necessary?

Reply:
Remember that in Swift, a subclass must override all designated initializers or none of them. Since `init(coder:)` has been overridden, `init(nibName:bundle:)` must be overridden as well.